pkg/apis/sinks/v1alpha1/job_sink_validation.go

```diff
@@ -33,5 +37,20 @@ func (sink *JobSinkSpec) Validate(ctx context.Context) *apis.FieldError {
 		return errs.Also(apis.ErrMissingOneOf("job"))
 	}
 
+	if sink.Job != nil {
+		job := sink.Job.DeepCopy()
+		job.Name = "temporary-job-name"
+		_, err := sinks.GetConfig(ctx).KubeClient.
+			BatchV1().
+			Jobs(apis.ParentMeta(ctx).Namespace).
+			Create(ctx, job, metav1.CreateOptions{
+				DryRun:          []string{metav1.DryRunAll},
+				FieldValidation: metav1.FieldValidationStrict,
+			})
+		if err != nil {
+			return apis.ErrGeneric(err.Error(), "job")
+		}
```

Comment:
Just so I understand correctly - we are validating the `Job` by doing a dry run of creating one with this spec?

Reply:
yes